### ISSUE TITLE
Fast server IP can be Domain? So I can reverse proxy.

### ISSUE THREAD

**Issue by shinebayar-g** [atech/postal#664]
Fast server: I guess it's used in clicking and tracking works. It tries to listen on port 80 on second static address, which means is it another web server besides Nginx? (Because I haven't got it running successfully yet, don't know). If it's another web-server and listen on port 80, can't we just reverse proxy this into Nginx? Because some cloud providers doesn't really have public IP address available for direct bind, which means they all are NATed through cloud network. Please take a look at following picture of GCP virtual server.
![image](https://user-images.githubusercontent.com/3091558/45918199-eda26100-beb5-11e8-99d9-9c9135e71e42.png)
As you see, there is no external IP address available to use. Which means I cannot simply bind to something like:
```
fast_server:
  enabled: true
  bind_address: 103.58.193.95
```

Another example: I created a fresh virtual server on DigitalOcean.com. But they only allow 1 network interface per server, which means I cannot have 2 external IP. Only 1. There is Floating IP option, that can be plugged into any server. But it doesn't add any external network interface, some kind of NAT also.

For these reason, I'm out of luck to use Clicking and tracking feature. Is there any workaround?

In [Documentation it says](https://github.com/atech/postal/wiki/Click-&-Open-Tracking#the-tracking-web-server): 

> Postal runs a separate web server process that listens on ports 5010 (for HTTP) and 5011 (for HTTPS).

Wait what? There is no sign of port 5010 or 5011 in config files. So where are them from? Hardcoded somewhere? If so should I proxy them into Nginx or nah?

**Comment by willpower232** [comment]
The config allows you to specify the ports you wish to use for the fast server: https://github.com/atech/postal/blob/master/config/postal.defaults.yml#L30-L31

So you could reverse proxy it.

The only problem would be that you would have to manually keep the nginx configuration updated with all the domains and create/link to the SSL certificates.

FYI DigitalOceans floating IP attaches to a second internal IP address (e.g. 10.16.0.6) so does actually give your server two public IP addresses.

**Comment by shinebayar-g** [comment]
> FYI DigitalOceans floating IP attaches to a second internal IP address (e.g. 10.16.0.6) so does actually give your server two public IP addresses.
Oh okay, I will try this. 

Some side note: Postal didn't work on GCP virtual server. It was giving some IPv6 related error when trying to send email, even after disabling IPv6 in both postal and kernel. So I went back to good old DigitalOcean vps.

**Comment by willpower232** [comment]
I believe there have been other issues around Googles desire that you don't send email from their servers so it already wasn't a perfect home for Postal, good to hammer it in though.

**Comment by shinebayar-g** [comment]
I just assigned Floating IP to my droplet and result is:
![image](https://user-images.githubusercontent.com/3091558/46010471-5e838c00-c0f5-11e8-87ea-267bcfa90ac1.png)
I tried to reboot, result was same.

When I try to bind to Floating IP:
![image](https://user-images.githubusercontent.com/3091558/46010521-8e329400-c0f5-11e8-9aad-2a3a3d21b500.png)

How can I achieve what you said above?

**Comment by willpower232** [comment]
It looks like you're missing the internal connection from your network configuration, you should see something like this

![image](https://user-images.githubusercontent.com/1619102/46010775-b4d3d900-c0bb-11e8-9247-6d4565303f68.png)

You should contact DigitalOcean support about enabling it on your droplet and I suspect you'll need to follow something like these instructions to enable your additional IP https://www.digitalocean.com/docs/networking/floating-ips/how-to/enable-old/

**Comment by shinebayar-g** [comment]
Hmm I see, I enabled private network following this tutorial https://www.digitalocean.com/docs/networking/private-networking/how-to/enable/.

Turned off Droplet and Enabled Private networking, followed the tutorial and here is the result.
![image](https://user-images.githubusercontent.com/3091558/46016365-29803500-c107-11e8-9f74-fcdc32124321.png)

And bind fast_server to this IP address.
![image](https://user-images.githubusercontent.com/3091558/46016461-78c66580-c107-11e8-811f-04e8c09b4e78.png)
red one is my main external IP address.
Blue one is new private network IP address.
Fast server is up. But when I access my Floating IP address in browser, I see this:
![image](https://user-images.githubusercontent.com/3091558/46016554-af9c7b80-c107-11e8-8ecd-6e7eb681be75.png)

However when I SSH into Floating IP, it connects to my Droplet without problem.

**Comment by willpower232** [comment]
Thats correct yes, if you enable click tracking for one of your domains and send some messages, you should be able to see whether they were opened or had links clicked now.
